src/main/java/org/eclipse/che/workspace/infrastructure/dummy/MyInternalRuntime.java

```diff
@@ -0,0 +1,184 @@
+import org.eclipse.che.api.workspace.server.spi.environment.InternalMachineConfig;
+import org.eclipse.che.api.workspace.shared.dto.event.MachineStatusEvent;
+import org.eclipse.che.api.workspace.shared.dto.event.ServerStatusEvent;
+import org.eclipse.che.dto.server.DtoFactory;
+import org.eclipse.che.workspace.infrastructure.dummy.bootstrapper.MyBootstrapper;
+import org.eclipse.che.workspace.infrastructure.dummy.bootstrapper.MyBootstrapperFactory;
+import org.eclipse.che.workspace.infrastructure.dummy.env.MyInternalEnvironment;
+
+/** @author Sergii Leshchenko */
+public class MyInternalRuntime extends InternalRuntime<MyRuntimeContext> {
+
+  private final MyBootstrapperFactory myBootstrapperFactory;
+  private final ServersCheckerFactory serverCheckerFactory;
+  private final EventService eventService;
+
+  @Inject
+  public MyInternalRuntime(
+      @Assisted MyRuntimeContext context,
+      @Assisted List<Warning> warnings,
+      URLRewriter urlRewriter,
+      MyBootstrapperFactory myBootstrapperFactory,
+      ServersCheckerFactory serverCheckerFactory,
+      EventService eventService) {
+    super(context, urlRewriter, warnings, false);
+    this.myBootstrapperFactory = myBootstrapperFactory;
+    this.serverCheckerFactory = serverCheckerFactory;
+    this.eventService = eventService;
+  }
+
+  @Override
+  protected Map<String, ? extends Machine> getInternalMachines() {
+    // Internal runtime should return list of machine with resolved servers here
+    return emptyMap();
+  }
+
+  @Override
+  protected void internalStart(Map<String, String> startOptions) throws InfrastructureException {
+    MyInternalEnvironment envToStart = getContext().getEnvironment();
+    try {
+      for (Entry<String, InternalMachineConfig> machineConfigEntry :
+          envToStart.getMachines().entrySet()) {
+        doStartMachine(machineConfigEntry);
+      }
+    } catch (Exception e) {
+      // clean up resources here.
+      try {
+        throw e;
+      } catch (InfrastructureException rethrow) {
+        throw rethrow;
+      } catch (Exception wrap) {
+        throw new InternalInfrastructureException(e.getMessage(), wrap);
+      }
+    }
+  }
+
+  private void doStartMachine(Entry<String, InternalMachineConfig> machineConfigEntry)
+      throws InfrastructureException, InterruptedException {
+    String machineName = machineConfigEntry.getKey();
+    InternalMachineConfig machineConfig = machineConfigEntry.getValue();
+
+    sendMachineStartingEvent(machineName);
+
+    try {
+      doStartMachine(machineName, machineConfig);
+
+      sendMachineRunningEvent(machineName);
+
+      doBootstrap(machineName, machineConfig);
+
+      doWaitServersRunning(machineName, machineConfig);
+    } catch (Exception e) {
+      sendMachineFailedEvent(machineName, e.getMessage());
+      throw e;
+    }
+  }
+
+  private void doStartMachine(String machineName, InternalMachineConfig machineConfig) {
+    //
```

Comment:
consider adding some comment, should not be empty  `//`

Reply:
+1
Added comment about machine starting in https://github.com/codenvy/che-infra-sample/pull/2